### ISSUE TITLE
Show compilation status at top of execution pane

### ISSUE THREAD

**Issue by vsajip** [integer32llc/rust-playground#576]
When iteratively working in "build" mode, the overall status of the compilation isn't immediately obvious in the "Execution" pane if there are many warnings shown at the top of the "Standard Error" sub-pane, and errors need to be scrolled down to. It would be useful to show overall status information (compilation either passed or failed) with e.g. a "Status" sub-pane above "Standard Error" which can be an overall success/failure indication, or else the "Standard Error" subpane title could incorporate the status of the compilation in the title itself (e.g. using "Standard Error (SUCCESS)" or "Standard Error (FAILURE)" as the title).

Alternatively, the execution pane could be automatically scrolled down to the bottom so that any errors are immediately visible to the user.

**Comment by shepmaster** [comment]
As an example, you are compiling this kind of code?

```rust
fn main() {
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    let a = 1;
    panic!("bang");
}
```

**Comment by shepmaster** [comment]
> It would be useful to show overall status information (compilation either passed or failed) with e.g. a "Status" sub-pane above "Standard Error" which can be an overall success/failure indication, or else the "Standard Error" subpane title could incorporate the status of the compilation in the title itself (e.g. using "Standard Error (SUCCESS)" or "Standard Error (FAILURE)" as the title).

I think either of these would be valid. I think that we return a success / failure boolean from the API call, but I'm not sure if it's actually stored in the state anywhere at the moment.

We could also have some alternate / additional status indicators, such as green text and/or an icon (e.g. ✅) for success cases.

> Alternatively, the execution pane could be automatically scrolled down to the bottom so that any errors are immediately visible to the user.

I don't think it's guaranteed that an error is the last thing, and I'm not a *super* fan of automatic scrolling in most cases, so I'd want to see how this looks and feels before committing to autoscrolling.

**Comment by vsajip** [comment]
> As an example, you are compiling this kind of code?

No. Just code which generates a few "unused variable"-type warnings (planning to use them later, honest!) on a laptop-sized screen, meaning that errors often (though not always) are not immediately visible. When working iteratively in the playground, it's not hard to go from a state of successful compilation (as determined by scrolling manually to the bottom of the execution pane) and then adding changes, scrolling down and checking, etc. until eventually you introduce a compilation error but don't realise it because the errors are down at the bottom, out of sight, and you didn't scroll down to check.

> We could also have some alternate / additional status indicators

Works for me.

>  I'm not a super fan of automatic scrolling in most cases, so I'd want to see how this looks and feels before committing to autoscrolling.

Fair enough. I don't mean an actual scrolling animation, just that the execution pane appears with the bottom-most part of the content visible.
